compatibility/compatibility-vec11to12/src/main/java/com/foursoft/harness/compatibility/vec11to12/wrapper/vec11to12/specification/Vec11To12StripeSpecificationWrapper.java

```diff
@@ -0,0 +1,98 @@
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ * =========================LICENSE_END==================================
+ */
+package com.foursoft.harness.compatibility.vec11to12.wrapper.vec11to12.specification;
+
+import com.foursoft.harness.compatibility.core.CompatibilityContext;
+import com.foursoft.harness.compatibility.core.wrapper.ReflectionBasedWrapper;
+import com.foursoft.harness.vec.v12x.VecCustomProperty;
+import com.foursoft.harness.vec.v12x.VecNumericalValue;
+import com.foursoft.harness.vec.v12x.VecNumericalValueProperty;
+
+import java.lang.reflect.Method;
+import java.util.List;
+
+/**
+ * Wrapper to wrap {@link com.foursoft.harness.vec.v113.VecStripeSpecification}
+ * to {@link com.foursoft.harness.vec.v12x.VecStripeSpecification}.
+ */
+public class Vec11To12StripeSpecificationWrapper extends ReflectionBasedWrapper {
+
+    private static final String GET_CUSTOM_PROPERTIES_METHOD_NAME = "getCustomProperties";
+    private static final String THICKNESS_CUSTOM_PROPERTY = "Thickness";
+
+    private VecNumericalValue thickness;
+
+    /**
+     * Creates this wrapper.
+     *
+     * @param context Context of the wrapper.
+     * @param target  Target object of the wrapper.
+     */
+    public Vec11To12StripeSpecificationWrapper(final CompatibilityContext context, final Object target) {
+        super(context, target);
+    }
+
+    @Override
+    protected Object wrapObject(final Object obj, final Method method, final Object[] allArguments) throws Throwable {
+        if ("getThickness".equals(method.getName())) {
+            return getThickness();
+        }
+        if ("setThickness".equals(method.getName())) {
+            return setThickness((VecNumericalValue) allArguments[0]);
+        }
+        if (GET_CUSTOM_PROPERTIES_METHOD_NAME.equals(method.getName())) {
+            return getCustomPropertiesWithoutThickness();
+        }
+
+        return super.wrapObject(obj, method, allArguments);
+    }
+
+    // The VEC 1.2.0 is not supposed to provide this CustomProperty anymore since it's an own field now.
+    private List<VecCustomProperty> getCustomPropertiesWithoutThickness() {
+        return wrapList(GET_CUSTOM_PROPERTIES_METHOD_NAME, VecCustomProperty.class).stream()
+                .filter(c -> !c.getPropertyType().equalsIgnoreCase(THICKNESS_CUSTOM_PROPERTY))
+                .toList();
+    }
+
+    private Object setThickness(final VecNumericalValue arg) {
+        thickness = arg;
+        return null;
+    }
+
+    private VecNumericalValue getThickness() {
+        if (thickness == null) {
+            thickness = wrapList(GET_CUSTOM_PROPERTIES_METHOD_NAME, VecCustomProperty.class).stream()
```

Comment:
You can maybe refactor this code and build a Utility function `getCustomPropertyNumericalValue(String methodName, String propertyType)`. But i'll leave it up to you, it is not a must have. You are doing a great job anyway.

Reply:
I was already thinking about making a utility function in general for accessing the custom properties. There should be more situations where the custom properties have to be accessed, even without a specific property type, I just didn't know where to put that method.

Do you have an idea? Else I'll think longer about this. Nevertheless, thank you again for your important input, I appreciate it very much!

Reply:
I was also wondering about that Issue and decided to put some braine juice into it. I think we could maybe have a Utility class for each Wrapper direction? That way the methods would not require the `11to12` or `12to11` string in each method, but the Utility classes would be `Vec12To11WrapperUtils` and `Vec11To12WrapperUtils`, also the corresponding prefix of the other Wrappers.  Im endefekt, there are no Common functions but each wrapping direction has its own implementations. We will achieve this because we are a great team.